src/tree/subscriptionTreeItem.ts

```diff
@@ -76,16 +84,79 @@ class SubscriptionTreeItem extends AzExtParentTreeItem implements SubscriptionTr
         const clusterResources = await getResources(
             this.sessionProvider,
             this.subscription.subscriptionId,
-            "Microsoft.ContainerService/managedClusters",
+            clusterResourceType,
         );
         if (failed(clusterResources)) {
             window.showErrorMessage(
                 `Failed to list clusters in subscription ${this.subscription.subscriptionId}: ${clusterResources.error}`,
             );
             return [];
         }
+        const fleetResources = await getResources(
+            this.sessionProvider,
+            this.subscription.subscriptionId,
+            fleetResourceType,
+        );
+        if (failed(fleetResources)) {
+            window.showErrorMessage(
+                `Failed to list fleets in subscription ${this.subscription.subscriptionId}: ${fleetResources.error}`,
+            );
+            return [];
+        }
+
+        // gather all members of each fleet found
+        const fleetToMembersMap = new Map<string, DefinedFleetMemberWithGroup[]>();
+        const memberPromises = fleetResources.result.map(async (f) => {
+            const members = await getFleetMembers(this.sessionProvider, f);
+            if (failed(members)) {
+                window.showErrorMessage(
+                    `Failed to list fleets in subscription ${this.subscription.subscriptionId}: ${members.error}`,
+                );
+                return null;
+            }
+            fleetToMembersMap.set(f.id, members.result);
+            return members.result;
+        });
+        await Promise.all(memberPromises);
+        const clusterToMemberMap = new Map<string, DefinedFleetMemberWithGroup>();
+        fleetToMembersMap.forEach((members) => {
+            members.forEach((member) => {
+                clusterToMemberMap.set(member.clusterResourceId, member);
+            });
+        });
 
-        return clusterResources.result.map((r) => createClusterTreeNode(this, this.subscriptionId, r));
+        const allResources = new Map<string, DefinedResourceWithGroup>();
+        fleetResources.result.concat(clusterResources.result).forEach((r) => {
+            allResources.set(r.id, r);
+        });
+
+        const fleetTreeNodes = new Map<string, FleetTreeNode>();
+
+        const clusterTreeItems = new Map<string, AzExtTreeItem>();
+        allResources.forEach((r) => {
+            if (r.type === "Microsoft.ContainerService/fleets") {
+                const fleetTreeItem = createFleetTreeNode(this, this.subscriptionId, r);
+                fleetTreeNodes.set(r.id, fleetTreeItem);
+                return fleetTreeItem;
+            } else if (r.type === "Microsoft.ContainerService/managedClusters") {
+                const member = clusterToMemberMap.get(r.id);
+                if (member) {
+                    const fleet = fleetTreeNodes.get(member.parentResourceId);
+                    fleet?.addCluster([r]);
```

Comment:
we are only adding clusters from current subscription as member to the fleet

we should add all members to the fleet, even if they aren't part of this subscription